### PR TITLE
Add TO Go cdns/name

### PR DESCRIPTION
**Pull request by rob05c**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1620/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1641/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1683/
Test PASSed.

**Comment by rob05c** [comment]
Duplicate of https://github.com/apache/incubator-trafficcontrol/pull/2102

**Comment by DylanVolz** [comment]
Tested, works as expected.

File: traffic_ops/traffic_ops_golang/routes.go [hunk 1]
```diff
@@ -107,6 +107,7 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
 		//CDN: CRUD
 		{1.1, http.MethodGet, `cdns/?(\.json)?$`, api.ReadHandler(cdn.GetRefType(), d.DB), auth.PrivLevelReadOnly, Authenticated, nil},
 		{1.1, http.MethodGet, `cdns/{id}$`, api.ReadHandler(cdn.GetRefType(), d.DB), auth.PrivLevelReadOnly, Authenticated, nil},
+		{1.1, http.MethodGet, `cdns/name/{name}$`, cdn.GetName(d.DB.DB), auth.PrivLevelReadOnly, Authenticated, nil},
```
Comment (DylanVolz) [inline traffic_ops/traffic_ops_golang/routes.go]: Note, this pr could only be this line:
`{1.1, http.MethodGet, `cdns/name/{name}$`, api.ReadHandler(cdn.GetRefType(), d.DB), auth.PrivLevelReadOnly, Authenticated, nil},`
without the other file.

Comment (mitchell852) [reply traffic_ops/traffic_ops_golang/routes.go]: @rob05c - can you change this to use the generic ReadHandler?

Comment (mitchell852) [inline traffic_ops/traffic_ops_golang/routes.go]: @rob05c - can you change this to use the generic ReadHandler like Dylan said? We need to stay within the parameters of the framework when possible imo.

**Comment by rob05c** [comment]
@mitchell852 Changed.

> We need to stay within the parameters of the framework when possible imo.

For the record, I don't agree, IMO we should be moving away from the interface/reflection "CRUDer" and toward helper functions, which are much easier to learn, read, and use.

But I made the change as requested, to expedite the Go rewrite.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1730/
Test PASSed.

**Comment by DylanVolz** [comment]
tested, works as expected. @mitchell852

**Pull request merged**